reactos/drivers/usb/usbport/usbport.h

```diff
@@ -0,0 +1,1138 @@
+#define USBPORT_FLAG_POWER_AND_CHIRP_OK 0x40000000
+#define USBPORT_FLAG_RH_INIT_CALLBACK  0x80000000
+
+/* PnP state Flags */
+#define USBPORT_PNP_STATE_NOT_INIT  0x00000001
+#define USBPORT_PNP_STATE_STARTED   0x00000002
+#define USBPORT_PNP_STATE_FAILED    0x00000004
+#define USBPORT_PNP_STATE_STOPPED   0x00000008
+
+/* Timer Flags */
+#define USBPORT_TMFLAG_TIMER_QUEUED       0x00000001
+#define USBPORT_TMFLAG_HC_SUSPENDED       0x00000002
+#define USBPORT_TMFLAG_HC_RESUME          0x00000004
+#define USBPORT_TMFLAG_RH_SUSPENDED       0x00000008
+#define USBPORT_TMFLAG_TIMER_STARTED      0x00000010
+#define USBPORT_TMFLAG_WAKE               0x00000020
+#define USBPORT_TMFLAG_IDLE_QUEUEITEM_ON  0x00000040
+
+/* Miniport Flags */
+#define USBPORT_MPFLAG_INTERRUPTS_ENABLED  0x00000001
+#define USBPORT_MPFLAG_SUSPENDED           0x00000002
+
+/* Device handle Flags (USBPORT_DEVICE_HANDLE) */
+#define DEVICE_HANDLE_FLAG_ROOTHUB     0x00000002
+#define DEVICE_HANDLE_FLAG_REMOVED     0x00000008
+#define DEVICE_HANDLE_FLAG_INITIALIZED 0x00000010
+
+/* Endpoint Flags (USBPORT_ENDPOINT) */
+#define ENDPOINT_FLAG_DMA_TYPE      0x00000001
+#define ENDPOINT_FLAG_ROOTHUB_EP0   0x00000002
+#define ENDPOINT_FLAG_NUKE          0x00000008
+#define ENDPOINT_FLAG_QUEUENE_EMPTY 0x00000010
+#define ENDPOINT_FLAG_ABORTING      0x00000020
+#define ENDPOINT_FLAG_IDLE          0x00000100
+#define ENDPOINT_FLAG_OPENED        0x00000200
+#define ENDPOINT_FLAG_CLOSED        0x00000400
+
+/* UsbdFlags Flags (URB) */
+#define USBD_FLAG_ALLOCATED_MDL      0x00000002
+#define USBD_FLAG_NOT_ISO_TRANSFER   0x00000010
+#define USBD_FLAG_ALLOCATED_TRANSFER 0x00000020
+
+/* Pipe handle Flags (USBPORT_PIPE_HANDLE) */
+#define PIPE_HANDLE_FLAG_CLOSED 0x00000001
+#define PIPE_HANDLE_FLAG_NULL_PACKET_SIZE 0x00000002
+
+/* Transfer Flags (USBPORT_TRANSFER) */
+#define TRANSFER_FLAG_CANCELED   0x00000001
+#define TRANSFER_FLAG_DMA_MAPPED 0x00000002
+#define TRANSFER_FLAG_SUBMITED   0x00000008
+#define TRANSFER_FLAG_ABORTED    0x00000010
+#define TRANSFER_FLAG_ISO        0x00000020
+#define TRANSFER_FLAG_DEVICE_GONE 0x00000080
+#define TRANSFER_FLAG_SPLITED    0x00000100
+
+extern KSPIN_LOCK USBPORT_SpinLock;
+extern LIST_ENTRY USBPORT_MiniPortDrivers;
+
+typedef USBD_STATUS* PUSBD_STATUS;
+
+typedef struct _USBPORT_COMMON_BUFFER_HEADER {
+  ULONG Length;
+  ULONG_PTR BaseVA;
+  PHYSICAL_ADDRESS LogicalAddress;
+  SIZE_T BufferLength;
+  ULONG_PTR VirtualAddress;
+  ULONG_PTR PhysicalAddress;
+} USBPORT_COMMON_BUFFER_HEADER, *PUSBPORT_COMMON_BUFFER_HEADER;
+
+typedef struct _USBPORT_ENDPOINT *PUSBPORT_ENDPOINT;
+
+typedef struct _USBPORT_PIPE_HANDLE {
+  ULONG Flags;
+  ULONG PipeFlags;
+  USB_ENDPOINT_DESCRIPTOR EndpointDescriptor;
+  UCHAR Padded;
+  PUSBPORT_ENDPOINT Endpoint;
+  LIST_ENTRY PipeLink;
+} USBPORT_PIPE_HANDLE, *PUSBPORT_PIPE_HANDLE;
+
+typedef struct _USBPORT_CONFIGURATION_HANDLE {
+  PUSB_CONFIGURATION_DESCRIPTOR ConfigurationDescriptor; // 00
```

Comment:
Remove all the comments about offsets. Add `C_ASSERT(FIELD_OFFSET(struct, field) == 0xXXX)` instead for those you consider critical